legal-api/src/legal_api/services/business_details_version.py

```diff
@@ -583,15 +584,17 @@ def party_role_revision_json(filing, party_role_revision, is_ia_or_after) -> dic
     @staticmethod
     def get_party_revision(filing, party_id) -> dict:
         """Consolidates all party changes up to the given transaction id."""
+        business_attr = LegalEntity if filing.legal_entity_id else AlternateName
         party_current = (
-            db.session.query(LegalEntity)
-            .filter(LegalEntity.change_filing_id == filing.id)
-            .filter(LegalEntity.id == party_id)
+            db.session.query(business_attr)
+            .filter(business_attr.change_filing_id == filing.id)
+            .filter(business_attr.id == party_id)
         )
 
-        party_version = history_cls(LegalEntity)
+        party_version = history_cls(business_attr)
         party_history = (
             db.session.query(party_version)
+            .options(defer(party_version.changed))  # TODO: remove this after update history_cls
```

Comment:
I don't think we need this todo comment?  how were you planning on changing the `history_cls`?

Also, have you confirm this actually works?  i.e. are history records properly included in the union?

Reply:
I thought the `changed` column could be removed in the future because it is optional. but, I was wrong.
I will remove the comments.

Reply:
Ok, but otherwise, you've confirmed using `defer` still brings back history records properly?

Reply:
Yes, I confirmed